packages/provider-elasticsearch/src/example-types/filters/tagsQuery.js

```diff
@@ -92,16 +98,45 @@ let buildResultQuery = (node, children = {}, groupsKey = 'tags') => ({
       },
       ...children,
     },
+    ...(keywordGenerations.length > 0 && {
+      keywordGenerations: {
+        filters: {
+          filters: F.compactObject(
+            F.keysToObject((word) => filter({ ...node, tags: [{ word }] }))(
+              keywordGenerations
+            )
+          ),
+        },
+        aggs: {
+          keyword_generation_sort: {
+            bucket_sort: {
+              sort: [{ _count: { order: 'desc' } }],
+            },
+          },
+        },
+      },
+    }),
   },
 })
 
-let result = async (node, search) => {
-  let aggs = buildResultQuery(node)
+let result = (generateKeywords) => async (node, search) => {
+  let keywords = node.generateKeywords
+    ? await generateKeywords(generationTagInputs(node.tags))
+    : []
+
+  let aggs = node.generateKeywords
+    ? buildResultQuery(node, {}, 'tags', keywords)
+    : buildResultQuery(node)
 
   return _.flow(
-    _.get('aggregations.tags.buckets'),
-    _.mapValues(_.get('doc_count')),
-    (results) => ({ results })
+    (results) => ({
+      tags: _.get('aggregations.tags.buckets', results),
```

Comment:
assuming you can omit zero count tags and there's nothing else on `aggregations`, you can unify the code path here and remove any mention of `tags` and `keywordGenerations`

Reply:
For example:

```js
  _.mapValues(
    (x) => F.compactMapValues('doc_count', x.buckets),
    (await search(aggs)).aggregations
  )
```

until we add `compactMapValues` to futil, you can inline it:

```js
let compactMapValues = _.flow(_.mapValues, F.compactObject)
``